packages/builder/src/pages/builder/app/[application]/settings/name-and-url.svelte

```diff
@@ -10,14 +10,22 @@
     Icon,
   } from "@budibase/bbui"
   import { AppStatus } from "constants"
-  import { overview } from "stores/portal"
+  import { store } from "builderStore"
+  import { apps } from "stores/portal"
   import UpdateAppModal from "components/start/UpdateAppModal.svelte"
+  import { API } from "api"
 
   let updatingModal
 
-  $: app = $overview.selectedApp
+  $: filteredApps = $apps.filter(app => app.devId == $store.appId)
+  $: app = filteredApps.length ? filteredApps[0] : {}
```

Comment:
NAB: but is there any reason you always pull the current app by searching the portal `apps` store rather than from the main builder `store` where we already have the app? We do always load the full list of apps anyway, but doing it this way means we're trying to keep both the apps store and the builder store updated with the current definition of the app being edited.

The side effect of doing it this way is that we fire the main `initialise` method of the builder store multiple times, and I'm worried that's going to keep creating new websocket connections. It also triggers a large cascade of other API requests which could be avoided if we just updated state in the frontend store.

NAB like I said so no need to refactor this unless my testing here does show we're creating multiple websocket connections, in which case we just might need a small refactor of that initialisation logic to ensure it kills it's previous connection if we re-initialise the store while already editing it.